### PR TITLE
Santa: Post distributed notification when showing block UI

### LINKED ISSUE #869: Listening to Events
How do I subscribe to events generated by Santa. 

For example, I have an app that would like to listen in when Santa blocks a binary and would like to trigger additional flow. I assume the Santa Notification is not customizable; would like to provide the user with additional info and next steps when a binary is blocked.

Is there a better way to do it that does not involve the SyncServer and all done locally?

### PR DESCRIPTION
**Pull request by russellhancox**
This allows users to supplement or replace the Santa UI, as well as trigger post-block tools if needed.

Fixes #869

### CODE REVIEW

File: Source/santa/SNTNotificationManager.m [hunk 1]
```diff
@@ -112,12 +114,49 @@ - (void)queueMessage:(SNTMessageWindowController *)pendingMsg {
 
   pendingMsg.delegate = self;
   [self.pendingNotifications addObject:pendingMsg];
+  [self postDistributedNotification:pendingMsg];
```
Comment (pmarkowsky) [inline Source/santa/SNTNotificationManager.m:117]: Should whether or not we send notifications be configurable?

File: Source/santa/SNTNotificationManager.m [hunk 2]
```diff
@@ -112,12 +114,49 @@ - (void)queueMessage:(SNTMessageWindowController *)pendingMsg {
...
 
   if (!self.currentWindowController) {
     [self showQueuedWindow];
   }
 }
 
+- (void)postDistributedNotification:(SNTMessageWindowController *)pendingMsg {
```
Comment (pmarkowsky) [inline Source/santa/SNTNotificationManager.m:124]: nit: Add a comment here describing what a distributed notification is.

**Review by pmarkowsky — CHANGES_REQUESTED** [review]
Can we add a test for this?

File: Source/santa/SNTNotificationManager.m (hunk 2 above)
Comment (russellhancox) [reply Source/santa/SNTNotificationManager.m:131]: Done

File: Source/santa/SNTNotificationManager.m (hunk 1 above)
Comment (russellhancox) [reply Source/santa/SNTNotificationManager.m:117]: I don't think so, the notifications are relatively cheap (esp. in the context of only sending them for de-duped blocked executions) and there's nothing sensitive in the notifications that can't easily be gleaned from other sources.

**Review by mlw — APPROVED** [review]

**Pull request merged**